File: rlab/reproducibility/lockfile.py
import shutil
from pathlib import Path


def capture_project_files(root: Path, destination: Path) -> tuple[Path, ...]:
    copied: list[Path] = []
    for name in ("pyproject.toml", "uv.lock", "poetry.lock", "requirements.txt", "lab.toml"):
        source = root / name
        if source.exists():
            target = destination / name
            target.parent.mkdir(parents=True, exist_ok=True)
            shutil.copy2(source, target)
            copied.append(target)
    return tuple(copied)

File: crates/rlab-core/src/reproducibility/lockfile.rs
use std::path::{Path, PathBuf};

pub fn find_lockfile(root: &Path) -> Option<PathBuf> {
    ["uv.lock", "Cargo.lock", "poetry.lock", "pdm.lock"]
        .iter()
        .map(|name| root.join(name))
        .find(|path| path.exists())
}
